orm/src/test/java/org/hibernate/search/test/query/dsl/DSLTest.java

```diff
@@ -890,6 +890,57 @@ public void testUseMatchQueryOnEmbeddedNumericIdCreatesTermQuery() throws Except
 		);
 	}
 
+	@Test
+	@TestForIssue(jiraKey = "HSEARCH-1793")
+	public void testQueryingForEntitiesFromDifferentIdSpaces() throws Exception {
+		QueryBuilder queryBuilder = fullTextSession.getSearchFactory()
+				.buildQueryBuilder().forEntity( EducationalInstitution.class ).get();
+
+		Query query = queryBuilder.bool()
```

Comment:
I prefer to have a test which makes sure the "whole thing" actually works. To assert on the created `Criteria` I'd have to use Byteman as you say or similar which I personally find quite cumbersome. And I still wouldn't know whether these criteria work as intended in the end, so I'd still need a test in the current form.

I'll add some more comments to the test itself.

> Maybe add a test for the exceptional case as well?

What exactly DYM by "exceptional case"?